integrations/access/accesslist/app.go

```diff
@@ -0,0 +1,251 @@
+/*
+Copyright 2023 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package accesslist
+
+import (
+	"context"
+	"time"
+
+	"github.com/gravitational/trace"
+	"github.com/jonboulle/clockwork"
+
+	alclient "github.com/gravitational/teleport/api/client/accesslist"
+	"github.com/gravitational/teleport/api/types"
+	"github.com/gravitational/teleport/api/types/accesslist"
+	"github.com/gravitational/teleport/api/utils/retryutils"
+	"github.com/gravitational/teleport/integrations/access/common"
+	"github.com/gravitational/teleport/integrations/access/common/recipient"
+	"github.com/gravitational/teleport/integrations/access/common/teleport"
+	"github.com/gravitational/teleport/integrations/lib"
+	"github.com/gravitational/teleport/integrations/lib/logger"
+	pd "github.com/gravitational/teleport/integrations/lib/plugindata"
+	"github.com/gravitational/teleport/lib/services"
+	"github.com/gravitational/teleport/lib/utils"
+	"github.com/gravitational/teleport/lib/utils/interval"
+)
+
+const (
+	// oneWeek is the number of hours in a week.
+	oneWeek = 24 * time.Hour * 7
+)
+
+type App[M MessagingBot] struct {
+	pluginName  string
+	apiClient   teleport.Client
+	accessLists services.AccessListsGetter
+	pluginData  *pd.CompareAndSwap[pd.AccessListNotificationData]
+	bot         M
+	job         lib.ServiceJob
+	clock       clockwork.Clock
+}
+
+func NewApp[M MessagingBot]() *App[M] {
```

Comment:
Maybe I missed it, but how do we actually configure whether the plugin will be receiving the notifications for access list reviews? Is it going to be "if the plugin is setup, notifications will be sent there", or is there going to be some configuration?

Reply:
This will be handled later, but basically individual plugins will need to set this up explicitly.

Reply:
Okay, so the idea here is that it has to be configured per plugin. You'll see references to `accessrequest.New` in mattermost, for example.